src/openforms/prefill/contrib/haalcentraal/plugin.py

```diff
@@ -5,61 +5,57 @@
 from django.utils.translation import gettext_lazy as _
 
 from glom import GlomError, glom
-from requests import RequestException
 from zds_client import ClientError
 
 from openforms.authentication.constants import AuthAttribute
 from openforms.plugins.exceptions import InvalidPluginConfiguration
 from openforms.pre_requests.clients import PreRequestClientContext
+from openforms.prefill.contrib.haalcentraal.constants import Attributes
 from openforms.submissions.models import Submission
 
 from ...base import BasePlugin
 from ...registry import register
-from .constants import Attributes
 from .models import HaalCentraalConfig
 
 logger = logging.getLogger(__name__)
 
 
+def get_config() -> HaalCentraalConfig | None:
+    config = HaalCentraalConfig.get_solo()
+    if not config.service:
+        logger.warning("No service defined for Haal Centraal prefill.")
+        return None
+    return config
+
+
 @register("haalcentraal")
 class HaalCentraalPrefill(BasePlugin):
     verbose_name = _("Haal Centraal")
     requires_auth = AuthAttribute.bsn
 
-    request_kwargs = dict(headers={"Accept": "application/hal+json"})
-
     @staticmethod
     def get_available_attributes() -> list[tuple[str, str]]:
-        return Attributes.choices
+        config = get_config()
+        if not config:
+            return Attributes.choices
```

Comment:
Because the service of haal centraal config is allowed to be empty this will make that get_available_attributes always returns the default value.